zerver/actions/streams.py

```diff
@@ -1054,13 +1059,20 @@ def get_non_subscribed_subs() -> list[tuple[UserProfile, Stream]]:
         return ([], not_subscribed)
 
     sub_ids_to_deactivate = [sub_info.sub.id for sub_info in subs_to_deactivate]
+
+    stream_to_subscribers: dict[Stream, set[UserProfile]] = defaultdict(set)
+    for sub_info in subs_to_deactivate:
+        stream_to_subscribers[sub_info.stream].add(sub_info.user)
+
     # We do all the database changes in a transaction to ensure
     # RealmAuditLog entries are atomically created when making changes.
     with transaction.atomic(savepoint=False):
         Subscription.objects.filter(
             id__in=sub_ids_to_deactivate,
         ).update(active=False)
 
+        update_stream_subscriber_count(stream_to_subscribers=stream_to_subscribers, count_sign=-1)
+
         # Log subscription activities in RealmAuditLog
         event_time = timezone_now()
         event_last_message_id = get_last_message_id()
```

Comment:
These actions changes seem incomplete; this number should also change when users are created, deactivated, or reactivated.